lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -24,6 +26,21 @@
     'lti',
 ]
 
+# exclude these in Metrics
+NON_STUDENT_ROLES = ['instructor', 'staff']
+
+
+def get_non_student_list(course_key):
+    """
+    Find all user_ids with instructor or staff roles in student_courseaccessrole table
+    """
+    non_students = CourseAccessRole.objects.select_related('user').filter(
+        course_id=course_key,
+        role__in=NON_STUDENT_ROLES,
+    ).values('user__id')
+
+    return [non_student['user__id'] for non_student in non_students]
```

Comment:
In fact I wonder if with NON_STUDENT_ROLES = ['staff'] we can remove the distinct.